### ISSUE TITLE
Domain Verification: Add a link back to WordPress.com on the Domain Contact Verification screen

### ISSUE THREAD

**Issue by apeatling** [Automattic/wp-calypso#37277]
Please watch the "Interacting with the Checklist and Domain Verification" testing video on this post: p58i-8d4-p2

* The “Domain Contact Verification” screen after clicking the email verification link is a dead end. There is no link back to WordPress.com. It does have instructions to close the window, but surely we can be more helpful than this.

**Comment by klimeryk** [comment]
Now that we have `wordpress.com/domains/manage` to manage all domains, we could quickly and safely redirect the user there.
Originally we did not provide such a link because: we did not have that page. But also because there's no guarantee the user is logged in - or that they are logged in as the owner of the domain :) These seem like edge cases, but they show up in customer interactions frequent enough to bubble up to the product team, so it's not _that_ infrequent. But `/domains/manage` should be a good compromise.
